### PR TITLE
Add eos_smoke integration tests

### PR DESCRIPTION
**Pull request by newswangerd**
##### SUMMARY

Adds  eos_smoke integration tests

covers:
module_utils:

    network.common.utils
    network.common.config
    network.eos

plugins:

    plugins.action.eos
    plugins. cliconf.eos
    plugins.terminal.eos

Coverage reports: https://newswangerd.github.io/eos/

##### ISSUE TYPE

 - Test Pull Request

##### ANSIBLE VERSION

```
ansible 2.6.0 (eos_smoke 3d82f1a6e7) last updated 2018/03/02 14:14:32 (GMT -400)
  config file = None
  configured module search path = [u'/Users/david/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/david/code/ansible/lib/ansible
  executable location = /Users/david/code/ansible/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/integration/targets/eos_smoke/tests/eapi/common_utils.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+- name: teardown
+  eos_config:
+    lines:
+    - no ip route 192.168.3.0/24 192.168.0.1
+    authorize: yes
+    provider: "{{ eapi }}"
+
+- debug: msg="END cli/common_utils.yaml on connection={{ ansible_connection }}"
+
+# FIXME: uncomment when https://github.com/ansible/ansible/issues/36919 is fixed
```
Comment (ganeshrn) [inline test/integration/targets/eos_smoke/tests/eapi/common_utils.yaml]: The issue is fixed, so as per comment the test should be uncommented.

**Review by ganeshrn — CHANGES_REQUESTED** [review]
Can you also add test cases related to become and unbecome

**Comment by newswangerd** [comment]
the unbecome/become tests will break until #38284 is fixed.

**Review by gundalow — CHANGES_REQUESTED** [review]
`become` & `unbecome` requires `connection: network_cli`.

There isn't an equivalent for `connection: local`, you can only `authorize`, no `unauthorize`

**Review by gundalow — APPROVED** [review]

**Review by ganeshrn — APPROVED** [review]

**Review by trishnaguha — APPROVED** [review]

**Comment by newswangerd** [comment]
backport: #38397

**Pull request merged**